File: testdata/fixtures/typescript/src/pkg/handler.ts
/**
 * Handler module - request handling
 */

import { Service } from "./service";
import { FormatOutput } from "../internal/util";

/**
 * Handler class for processing requests
 */
export class Handler {
  private service: Service;

  constructor(service: Service) {
    this.service = service;
  }

  /**
   * Handles a single request
   * @param input The input to process
   * @returns The formatted output
   */
  handle(input: string): string {
    const result = this.service.process(input);
    return FormatOutput(result);
  }

  /**
   * Handles a batch of requests
   * @param inputs The inputs to process
   * @returns Array of results
   */
  handleBatch(inputs: string[]): string[] {
    return inputs.map((input) => this.handle(input));
  }
}

/**
 * Factory function for creating handlers
 * @param service The service to use
 * @returns New Handler instance
 */
export function newHandler(service: Service): Handler {
  return new Handler(service);
}

/**
 * Standalone handler function (for disambiguation testing)
 * Demonstrates handler() function vs Handler class
 * @param input The input to process
 * @returns The processed string
 */
export function handler(input: string): string {
  return `handled: ${input}`;
}

File: testdata/fixtures/go/pkg/handler.go
// Package pkg provides HTTP handling and business logic.
package pkg

import "fixture/internal"

// Handler is a type that handles HTTP requests (disambiguation: same name as main.Handler function).
type Handler struct {
	service Service
}

// NewHandler creates a new Handler with the given service.
func NewHandler(svc Service) *Handler {
	return &Handler{service: svc}
}

// Handle processes a request using the service.
// Call chain: Handle -> service.Process (interface call)
func (h *Handler) Handle(input string) string {
	result := h.service.Process(input)
	return internal.FormatOutput(result)
}

// HandleBatch processes multiple inputs.
func (h *Handler) HandleBatch(inputs []string) []string {
	results := make([]string, len(inputs))
	for i, input := range inputs {
		results[i] = h.Handle(input)
	}
	return results
}
